doc/build/dts/howtos.rst

```diff
@@ -247,6 +247,8 @@ use as devicetree overlays:
 #. Otherwise, if :file:`<BOARD>.overlay` exists, it will be used, and the build
    system will stop looking for more files.
 #. Otherwise, if :file:`app.overlay` exists, it will be used.
+#. If the file :file:`socs/<SOC>_<BOARD_QUALIFIERS>.overlay` exists, it will be
+   included as an additional file
```

Comment:
This mess up the docs, and makes it harder to follow the flow.

Because when reading step by step in existing doc, you can terminate the flow at multiple occasions:
https://github.com/zephyrproject-rtos/zephyr/blob/792fd57d96327dee3d7477479489ebd9fc42f907/doc/build/dts/howtos.rst?plain=1#L245-L246
and:
https://github.com/zephyrproject-rtos/zephyr/blob/792fd57d96327dee3d7477479489ebd9fc42f907/doc/build/dts/howtos.rst?plain=1#L247-L248

but those sentences are no longer true, because of this proposed `If the file` at the bottom of those steps.

Probably we should follow the existing overlay file structure and only used the qualifier overlay in addition to board overlay (and only board overlays from `boards/` folder), not app.overlay.

See also: https://github.com/zephyrproject-rtos/zephyr/pull/70418#discussion_r1580748053

Reply:
Have updated code and docs